src/libponyrt/asio/event.c

```diff
@@ -68,6 +68,14 @@ PONY_API int pony_asio_event_fd(asio_event_t* ev)
   return ev->fd;
 }
 
+PONY_API bool pony_asio_event_get_disposable(asio_event_t* ev)
+{
+  if(ev == NULL)
+    return false;
+
+  return (ev->flags & ASIO_DISPOSABLE) != 0;
```

Comment:
This should be `return (ev->flags == ASIO_DISPOSABLE);` because `ASIO_DISPOSABLE` has a value of `0` as defined in the enum in `asio.h` and because when `ev->flags` is assigned `ASIO_DISPOSABLE` it is assigned as `ev->flags = ASIO_DISPOSABLE` unsetting any other flags previously assigned.

Reply:
Ouch, yes, I'll fix that.